skyportal/handlers/api/tag.py

```diff
@@ -0,0 +1,238 @@
+import re
+
+from sqlalchemy import func
+
+from baselayer.app.access import auth_or_token
+
+from ...models import Obj, ObjTag, ObjTagOption
+from ..base import BaseHandler
+
+
+class ObjTagOptionHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        with self.Session() as session:
+            tags = session.scalars(ObjTagOption.select(session.user_or_token)).all()
+            return self.success(data=tags)
+
+    @auth_or_token
+    def post(self):
+        data = self.get_json()
+        name = data.get("name")
+
+        if not name or not isinstance(name, str):
+            return self.error("`name` must be provided as a non-empty string")
+
+        if not re.fullmatch(r"[A-Za-z0-9]+", name):
+            return self.error(
+                "`name` must contain only letters and numbers (no spaces, underscores, or special characters)",
+                status=400,
+            )
+
+        with self.Session() as session:
+            existing_tag = session.scalars(
+                ObjTagOption.select(session.user_or_token).where(
+                    func.lower(ObjTagOption.name) == name.lower()
+                )
+            ).first()
+
+            if existing_tag:
+                return self.error(
+                    f"Tag '{name}' already exists (case-insensitive match)",
+                    status=409,
+                )
+
+            new_tag = ObjTagOption(name=name)
+            session.add(new_tag)
+            session.commit()
+
+            return self.success(new_tag)
+
+    @auth_or_token
+    def patch(self, tag_id):
```

Comment:
I suggest checking that you can cast the tag_id to an int, to make sure it's a valid ID. This or in `skyportal_app_server.py` ideally your regex would not just match anything like it does now, but also numbers. I suggest doing that actually, dealing with that in the route is best. That said still a good idea to cast to int here otherwise the queries might not work as expected.

Reply:
that comment is valid for all the endpoints btw